cxx/util_distribution_variant.hh

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024
+// See LICENSE.txt
+
+// This file collects classes/functions that depend on the set of distribution
+// subclasses and should be updated when a new subclass is added.
+
+#pragma once
+
+#include <map>
+#include <random>
+#include <string>
+#include <variant>
+#include <vector>
+
+enum class DistributionEnum { bernoulli, bigram, categorical, normal };
+
+struct DistributionSpec {
+  DistributionEnum distribution;
+  std::map<std::string, std::string> distribution_args = {};
+};
+
+class BetaBernoulli;
+class Bigram;
+class DirichletCategorical;
+class Normal;
+class Domain;
+template <typename DistributionType>
+class Relation;
+
+// Set of all distribution sample types.
+using ObservationVariant = std::variant<double, int, std::string>;
```

Comment:
Could you add a
// TODO(thomaswc):  Change BetaBernoulli to use bool's instead of double's.
?

Reply:
Done, in the BetaBernoulli class. This was on my list of things to get to also.